R/to_xml.R

```diff
@@ -87,6 +87,19 @@ transform_block <- function(code_block){
   # This prevents partial code blocks that are still apparently valid: ```{r, }
   info <- info[names(info) != ""]
 
+  names(info)[!(names(info) %in% c("language", "name"))] <-
+    paste0(names(info)[!(names(info) %in% c("language", "name"))], "-outchunk")
+
+  code <- strsplit(xml2::xml_text(code_block), "\n")[[1]]
+  inchunk_info <- knitr::partition_chunk(info[["language"]], code)
+  xml2::xml_text(code_block) <- paste0(inchunk_info$code, "\n")
```

Comment:
do not forget this newline :fire:

Reply:
what do you mean? I think it makes sense because the user might change the options via the XML so we need to write the in-chunk options again.

Reply:
That's a good point. I was thinking a situation where the user may want to know how many lines of code they have including options. But you are right. It would be more complex to retain the chunk option formatting in the first place.

Reply:
should this go into a feature where we transform code chunks info into a tibble? I mean adding a column with number of code lines.